### ISSUE TITLE
Mozilla will shortly be requiring MANDATORY signing of addons

### ISSUE THREAD

**Issue by affinityv** [RequestPolicyContinued/requestpolicy#600]
Before long, Mozilla will not allow the installation of addons that are not signed by them.

They will not make exceptions to this change.

Therefore, the only way we can install this addon is if it gets signed by Mozilla or we stop using the standard Firefox download.

Are you going to make sure this addon gets signed by Mozilla to avoid these issues?

**Comment by ansell** [comment]
Reference for this is:

https://blog.mozilla.org/addons/2015/02/10/extension-signing-safer-experience/

With enough convincing they may back down on it, as it is a silly idea for technical users, but in the meantime, there may be issues with it.

**Comment by myrdd** [comment]
Thanks for the info. Yes, I'll make sure that RequestPolicy will work with firefox. I think this is solved as soon as RP is on AMO, see https://github.com/RequestPolicyContinued/requestpolicy/issues/543

I'm wondering if extensions can be signed multiple times, so that my own signature remains – see https://github.com/RequestPolicyContinued/requestpolicy/issues/465

@ansell on the blog post they write *„Installation of unsigned extensions will still be possible on Nightly and Developer Edition“* so you might want to try Dev Edition?

**Comment by ansell** [comment]
@myrdd I imagine that when it comes into play I will be lobbying Ubuntu to remove the restriction in their version, for whatever that could be worth.

If necessary I could try their alternatives but it still doesn't seem right to never be able to test against the actual browser that users will be running on, and forcing all beta users to change their browser.

**Comment by nodiscc** [comment]
Interesting discussion regarding this at https://github.com/nylira/prism-break/issues/1235. It seems there will be no way around this (no about:config switch to disable it). This is planned for Firefox 39 which will be the stable release on **2015-06-30** (https://wiki.mozilla.org/RapidRelease/Calendar#Future_branch_dates). So RPC needs to be uploaded to AMO and signed by Mozilla at this date to continue working on the stable release.

I'd also like if we could keep @myrdd's signature on the XPI.

**Comment by myrdd** [comment]
@nodiscc that's correct. We have to be on AMO even earlier though, due to https://github.com/RequestPolicyContinued/requestpolicy/issues/525#issuecomment-71155275

**Comment by ldgbc** [comment]
Here are some resource relating to Add-on Signing:
https://blog.mozilla.org/addons/2015/04/15/the-case-for-extension-signing/
https://wiki.mozilla.org/Addons/Extension_Signing

Some add-on I used that recently have signing as of  2015-04-24; 
DownThemAll! AntiContainer	1.3.1-signed
NoScript	2.6.9.22.1-signed

Look like any updated version after this period will be signed add-on.

**Comment by myrdd** [comment]
I've just re-checked our plan. I've figured out it might be a little complicated…

Fx40 will introduce the requirement for addons to be signed. Fx40 is scheduled to 2015-08-11, so I'm confident the AMO version (essentially #609) will be finished until then.

However, I don't know what happens if an addon is still in the Review Queue, will it be impossible to install that addon? It's quite probable that RPC 0.5.29 will still be in the Review Queue in August.
I've uploaded it on May 20, 2015. The [status report](https://blog.mozilla.org/addons/2015/05/20/add-ons-update-65/) says *ten weeks* (!) in the Review Queue aren't unusual. Ten weeks would mean *end of july*, but I think it might be even later, maybe even after Aug 08 (Fx40 release date).

So what to do? Maybe what we need to do is using „our own“ update process via `updateURL`. This [is already the case for `v1.0.beta*`](https://github.com/RequestPolicyContinued/requestpolicy/blob/v1.0.beta9.3/src/install.rdf#L22), but maybe it's necessary for `0.5.*` as well.

When having our own „channels“, then we simply need Mozilla to sign the addon, nothing more. Afaict the Review Process is only necessary for AMO-offering, *not* for extension signing only.

The „feature“ to get an addon signed will be possible starting next week, according to [this blog post](https://blog.mozilla.org/addons/2015/05/27/update-signing-new-developer-agreement/). I will then test if it works.

---------------------------------

So in **summary**, my current plan is:

* wait until extension signing is possible (announced for next week)
* try if RPC 0.5.29 can be signed
  * if it fails for the first time, repeat until it works.
* as soon as `0.5.*` can be signed…
  * … create a 0.5 version with a custom`updateURL`, e.g. 0.5.30
    * let it sign as well
    * release it on github
    * offer this version on the website
  * … get RPC `1.0.beta10` signed
    * first finish #609 
    * then get it signed
    * release RPC `1.0.beta10`
    * *after that, keep the 1.0.beta „update channel“ (`updateURL`) as long as appropriate*
* As soon as the AMO review of RPC 0.5.* is finished, that version will be visible on AMO. Then …
  * the 0.5 version with a custom `updateURL` isn't necessary anymore
  * it will be possible to create a „beta“ channel on AMO. (it's not possible for unreviewed addons!) We could uset that beta channel for the 1.0 beta releases.

**Comment by DJ-Leith** [comment]
In #661 **myrdd** said:

> Yeah, in fact we know it already for many months. We even pushed back the "incompatibility" from Fx36 to Fx38. See #525. However, what we did *not* expect is that Firefox disables the addon automatically (!) on upgrade.

I have just found this on AMO!

**RequestPolicy 0.5.28.1-signed**
By Justin Samuel
Last Updated 2013-07-30
Homepage https://www.requestpolicy.com

Using a fresh Profile, on Fx 38.0.1 (Release), I can install it.
It would have been wonderful if RP 0.5.28 had NOT been 'disabled by Mozilla' as Users updated their Firefox from Fx 37 to Fx 38 around 2015-05-12!

So, I speculate, Mozilla are automatically singing all (or many) old addons.

I suppose one good 'side effect' has been to get more folk using RPC.
The downside is, I fear, many may have 'given up' on RP/RPC in the last few weeks.

**Comment by jdgalt** [comment]
There is (or at least used to be) a setting in about:config that disables Firefox' blacklist of unapproved add-ons.  If necessary, we may want to publish it.

**Comment by bkimmett** [comment]
Just checking in. What's the status of the AMO version?

**Comment by myrdd** [comment]
@DJ-Leith signing of addons has nothing to do with the "incompatibility" of RP 0.5.28.

---

@jdgalt If you mean `extensions.blocklist.enable`, I would rather link to [RPC 0.5.29](https://addons.mozilla.org/en-US/firefox/addon/requestpolicy-continued/).

---

@bkimmett Uploading RPC to AMO is tracked in issue https://github.com/RequestPolicyContinued/requestpolicy/issues/543, see its first post. If you meant status of a signed version: it mainly depends on #609.

In the last few days I've been working on the unit testing environment (not pushed yet), see #641. It has nothing to do with AMO and signing, but it's important as well. I'm going to write unit tests that ensure the user will be warned when both RP and RPC are activated, as that might cause problems.

I think it's a good idea to publish a mozilla-signed release soon, so I'll focus on that.

**Comment by DJ-Leith** [comment]
> @DJ-Leith signing of addons has nothing to do with the "incompatibility" of RP 0.5.28.

@myrdd I agree, the mechanisms are different (blocklisting is another *different* mechanism).
The effect / result - '**disable** the Extension' - is my concern.

We unexpectedly found that Users who had RP 0.5.28 had their 'RP disabled' when their Firefox became Fx&nbsp;38.

In **Firefox 38 broke RequestPolicy 0.5** #661 myrdd said:
> Yeah, in fact we know it already for many months. We even pushed back the "incompatibility" from
> Fx36 to Fx38. See #525. However, what we did not expect is that Firefox disables the addon
> automatically (!) on upgrade.

My reading of "signing" [1] is that Mozilla intend to disable all non-singned by Mozilla Extensions (for Fx&nbsp;Release and Fx&nbsp;Beta).
So, even the RequestPolicy 1.0.beta9.3 will be 'disabled by Mozilla', on Fx&nbsp;Release and Fx&nbsp;Beta (but not Fx&nbsp;Dev&nbsp;Ed or Fx&nbsp;Nightly).
Mozilla do NOT want any way to avoid this 'block'.

I think your plan [above]
(https://github.com/RequestPolicyContinued/requestpolicy/issues/600#issuecomment-106958748) is good.

> I think it's a good idea to publish a mozilla-signed release soon, so I'll focus on that.

In view of the fact that the 'review queue' is slow, would it be a good idea to get <code>1.0.beta9.3</code> signed by Mozilla?
In other words: I am suggesting - don't wait for <code>1.0.beta10</code>.


[1]
https://wiki.mozilla.org/Addons/Extension_Signing

**Comment by myrdd** [comment]
> In view of the fact that the 'review queue' is slow, would it be a good idea to get 1.0.beta9.3 signed by Mozilla?

Sadly that's not possible. RPC 1.0.beta9.3 has the same extension ID as [RP 0.5.28 on AMO](https://addons.mozilla.org/en-US/firefox/addon/requestpolicy/). Therefore this issue depends on #609.

> In other words: I am suggesting - don't wait for 1.0.beta10.

The `beta10` milstone contains only "changing the extension ID" and a translation. No worries, I'll do my best to prevent RPC being disabled.

**Comment by myrdd** [comment]
FYI: Looks like Mozilla has changed the timeline for extension signing: https://wiki.mozilla.org/index.php?title=Addons%2FExtension_Signing&action=historysubmit&diff=1081715&oldid=1079578

**Comment by myrdd** [comment]
Good news! The addon has been signed! I also managed to get rid of all warnings of the AMO validator so that RequestPolicy can be signed automatically.

> I'd also like if we could keep @myrdd's signature on the XPI.

I don't think that's possible. In the source code of the AMO validator you can read:

```
            description="Add-ons which are already signed will be re-signed "
                        "when published on AMO. This will replace any "
                        "existing signatures on the add-on.")
```

https://github.com/mozilla/amo-validator/blob/67f72f2cdc68da76e91c1d5c9ba5aa51c958ed4d/validator/testcases/content.py#L283

**Comment by Diapolo** [comment]
So 41.0b1 disabled my installed RequestPolicy Continued 1.0.beta10 and can't find an update. Is beta 11 available for download yet?

**Comment by danielmotaleite** [comment]
No, looks like the 1.0.beta10-signed version is still not released... 1.0.beta11 will probably take a little longer

**Comment by myrdd** [comment]
Thanks for the reminder, I'll release the signed version today.

@Diapolo I'm sorry RP has been disabled. Yes, beta11 takes a little longer.
